std/http/headers.zig

```diff
@@ -102,9 +102,19 @@ test "HeaderEntry" {
     testing.expectEqualSlices(u8, "x", e.value);
 }
 
+fn stringEql(a: []const u8, b: []const u8) bool {
+    if (a.len != b.len) return false;
+    if (a.ptr == b.ptr) return true;
+    return mem.compare(u8, a, b) == .Equal;
+}
+
+fn stringHash(s: []const u8) u32 {
+    return @truncate(u32, std.hash.wyhash(s, 0));
+}
+
 const HeaderList = std.ArrayList(HeaderEntry);
 const HeaderIndexList = std.ArrayList(usize);
-const HeaderIndex = std.AutoHashMap([]const u8, HeaderIndexList);
+const HeaderIndex = std.HashMap([]const u8, HeaderIndexList, stringHash, stringEql);
```

Comment:
Why hard-code this in here? (why not use AutoHashMap directly any more?)

Reply:
Because this code relied on the old behaviour of `autoHash` and `autoEql` that followed pointers (slice in this case), see https://github.com/ziglang/zig/pull/2797/files#diff-4203791c602f5595906b6d22157d6e50R540.

I changed the default behaviour to `not` follow pointers as that's what you would expect from generic code, if the map needs specific interpretation of the key/values it needs to be specified.

Reply:
Ah right. I didn't notice that change.
Is it worth adding a helper/utility HashMap that does operate on the contents of slices? I imagine http.headers isn't going to be the only one wanting to do so.

Reply:
That's a thing to consider. It can make sense to have a specialized hashmap for strings, since it's a really frequent non-default need.